src/java.base/macosx/native/libosxsecurity/KeystoreImpl.m

```diff
@@ -492,11 +493,6 @@ static void addCertificatesToKeystore(JNIEnv *env, jobject keyStore,
                 goto errOut;
             }
 
-            // Only add certificates with trust settings
-            if (inputTrust == NULL) {
```

Comment:
From what I can tell non root certificates that do not have explicit trust settings do not show up in `SecTrustSettingsCopyTrustSettings`, docs appear to be https://developer.apple.com/documentation/security/sectrustsettingscopytrustsettings(_:_:_:) but not very clear.

----

I need to test that the certificate is still chained to a root and not trusted as a root.

Reply:
Ok this isn't working properly 😢 

1. ⛔ Fails: Marking the certificate as OS default (which for CA certs is trust: false) - with an intermediate (I haven't tried without yet)
2. ⛔ Fails: Removing the root but leaving the intermediate

Reply:
Interesting for root certificate `SecTrustSettingsCopyTrustSettings` returns:

* -25300 (not found) when trust policy is `Use System Defaults`
* 0 and a `kSecTrustSettingsResult` value of 3 when set to Never Trust
* 0 and a `kSecTrustSettingsResult` value of 1 when set to Always Trust

Reply:
With https://github.com/openjdk/jdk/pull/22911/commits/5102dade13f44dedd887920c407158e7d189947b

Case 2. works again.

(i.e. the basic case which previously worked with a self-signed root and no intermediate)

Case 1 and 3 are still failing, I'll have a think on Monday but may need a pointer

Reply:
With https://github.com/openjdk/jdk/pull/22911/commits/0052cd0380b4949b9af689eae660cf3defa5e7d0 all cases are working